video/out/gpu/lcms.c

```diff
@@ -522,3 +482,37 @@ bool gl_lcms_get_lut3d(struct gl_lcms *p, struct lut3d **result_lut3d,
 }
 
 #endif
+
+static int validate_3dlut_size_opt(struct mp_log *log, const m_option_t *opt,
+                                   struct bstr name, const char **value)
+{
+    struct bstr param = bstr0(*value);
+    int p1, p2, p3;
+    char s[20];
+    snprintf(s, sizeof(s), "%.*s", BSTR_P(param));
+    return gl_parse_3dlut_size(s, &p1, &p2, &p3);
```

Comment:
unrelated but what is happening in this function?
all this work is equivalent to just
```c
{
    int p1, p2, p3;
    return gl_parse_3dlut_size(*value, &p1, &p2, &p3);
}
```

Reply:
I have no clue. I think you're right.